products/compute/api.yaml

```diff
@@ -935,6 +935,154 @@ objects:
+        description: |
+           The list of URLs to the HttpHealthCheck or HttpsHealthCheck resource
+           for health checking this BackendService. Currently at most one health
+           check can be specified, and a health check is required.
+
+           For internal load balancing, a URL to a HealthCheck resource must be
+           specified instead.
+      - !ruby/object:Api::Type::Array
+        name: 'backends'
+        description: |
+          The list of backends that serve this RegionBackendService.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'description'
+              description: |
+                An optional description of this resource.
+                Provide this property when you create the resource.
+            - !ruby/object:Api::Type::String
+              name: 'group'
+              description: |
+                The fully-qualified URL of an Instance Group. This defines the list
+                of instances that serve traffic. Member virtual machine
+                instances from each instance group must live in the same zone as
+                the instance group itself. No two backends in a backend service
+                are allowed to use same Instance Group resource.
+
+                Note that you must specify an Instance Group
+                resource using the fully-qualified URL, rather than a
+                partial URL.
+
+                The instance group must be within the same region as the BackendService.
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          An optional description of this resource.
+      - !ruby/object:Api::Type::Fingerprint
+        name: 'fingerprint'
+        output: true
+        description: |
+          Fingerprint of this resource. A hash of the contents stored in this
+          object. This field is used in optimistic locking.
+      - !ruby/object:Api::Type::Enum
+        name: 'protocol'
+        description: |
+          The protocol this BackendService uses to communicate with backends.
+
+          The possible values are TCP and UDP, and the default is TCP.
+        # This is removed to avoid breaking terraform, as default values cannot be
+        # unspecified. Providers should include this as needed via overrides
+        # default_value: :TCP
+        values:
+          - :TCP
+          - :UDP
+      - !ruby/object:Api::Type::Enum
+        name: 'sessionAffinity'
+        description: |
+          Type of session affinity to use. The default is NONE.
+
+          Can be NONE, CLIENT_IP, CLIENT_IP_PROTO, or CLIENT_IP_PORT_PROTO.
+
+          When the protocol is UDP, this field is not used.
+        values:
+          - :NONE
+          - :CLIENT_IP
+          - :CLIENT_IP_PROTO
+          - :CLIENT_IP_PORT_PROTO
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'region'
```

Comment:
This needs to have an override for `default_from_api` so it's `Computed` + `Optional`

Reply:
Why would region be computed and optional? It's required for a regional backend service

Reply:
We can pull the region from a few different places, so it ends up being optional: https://github.com/terraform-providers/terraform-provider-google/blob/5ce5686d4997d4b422e63f5e295bc9763f2a997d/google/utils.go#L39-L46

Reply:
Like this?
```
      region: !ruby/object:Overrides::Terraform::PropertyOverride
        default_from_api: true
        required: false
```

Reply:
Yep! There are some other extra overrides too that make it work a little better as well, see https://github.com/GoogleCloudPlatform/magic-modules/blob/16e6fe42ebdb772e323e3fab7dad503e1de8ea8d/products/compute/terraform.yaml#L52-L58